src/coreclr/vm/assembly.cpp

```diff
@@ -686,6 +686,21 @@ Assembly *Assembly::CreateDynamic(AppDomain *pDomain, ICLRPrivBinder* pBinderCon
             pAssem->SetIsTenured();
             pRetVal = pAssem;
         }
+
+        {
+            // We should be to create dependency link between LoaderAllocator of the dynamic assembly
+            // and the LoaderAllocator of the ALC if they are both unloadable.
```

Comment:
You don't need to do any of the collectibility checks here. The EnsureReference does these and creates the reference only if both are collectible.

Reply:
A nit - you could also put this code right after the `createdNewAssemblyLoaderAllocator = TRUE;`

Reply:
Removed unnecessary checks and moved the code to the section with `createdNewAssemblyLoaderAllocator = TRUE`;